docs: how to run java tests against a setup

A possible related issue https://github.com/JanssenProject/jans/issues/2876


We need a clear documentation added detailing how to run tests against a setup. 

General outline:
1. Run linux setup with -t
2. prepare env and copy any profiles
3. trigger `mvn clean compile install`

Instructions merged using the PR above are good to go based on tests on local lxd. **They give same results as tests run on Jenkins builds.**

In the most recent run on the local setup done using documented instructions, the result was
```
Tests run: 1839, Failures: 251, Errors: 0, Skipped: 9
```

241 of the 251 failures were due to Errors like the first error shown below. And then there were other errors too as listed. But all these tests were passing when run individually. So, actually zero tests failed. All these errors (especially the timeout errors) seem to be happening due to possible performance issues in my local lxd container as tests are passing individually.

Error (5) is not due to timeout but it is failing on the Jenkins build as well. Build #10082.

Types of erros in the logs:

(1)

BackchannelAuthenticationPingMode.idTokenHintAlgA256KWEncA256GCM:2588->BaseTest.authenticateResourceOwnerAndGrantAccess:441->BaseTest.authenticateResourceOwnerAndGrantAccess:450->BaseTest.authenticateResourceOwnerAndGrantAccess:460->BaseTest.authenticateResourceOwnerAndGrantAccess:480->BaseTest.acceptAuthorization:608->BaseTest.waitForRequredElementLoad:587 » Timeout Expected condition failed: waiting for io.jans.as.client.BaseTest$$Lambda$302/0x00000001007dbc40@131b8735 (tried for 30 second(s) with 1000 milliseconds interval)

(2)

PersistClientAuthorizationsHttpTest.doNotPersistAuthorizationWhenPreAuthorized:461 The scope is null expected object to not be null

(3)

AuthorizationResponseModeFragmentJwtResponseTypeTokenEncryptedHttpTest.testAlgA128KWEncA128GCM:61->BaseTest.authorizationRequest:1225->BaseTest.authoriz    ationRequest:1241 expected object to not be null

(4)

AuthorizeRestWebServiceHttpTest.requestAuthorizationDenyAccessThenGrantAccess:2189->BaseTest.authenticateResourceOwnerAndDenyAccess:708 » NoSuchElement     Unable to locate element with ID: 'authorizeForm:doNotAllowButton'

(5)

[ERROR]   RegistrationWithSoftwareStatementTest.requestClientAssociate2:245 Unexpected response code: {
    "error_description": "The software_statement is malformed or invalid. CorrelationId: dcd4cc5e-22f1-45c9-b2ac-6b6000ce364b",
    "error": "invalid_software_statement"
} expected [201] but found [400]

--- a/docs/contribute/development.md
+++ b/docs/contribute/development.md
@@ -284,3 +284,12 @@ setting up a profile for the server module.
 
 1. Under `jans-scim/client/profiles` directory, create a new directory and name it `janssen2.op.io`
 2. Copy the contents of `scim-client/client` directory into the newly created `janssen2.op.io` directory
+
+Post this, remove the `scim-client` directory.
+
+Now as the profile is setup, to build the `jans-scim` module and run tests, 
+use the command below:
+
+  ```
+  mvn -Dcfg=janssen2.op.io test
+  ```

--- a/docs/contribute/development.md
+++ b/docs/contribute/development.md
@@ -192,7 +192,7 @@ Follow the steps below to configure the profile for the client module. The same
 steps should be followed for
 setting up a profile for the server module.
 
-1. Under `jans-auth-server/client/profile` module, create a new directory and name it `janssen2.op.io`
+1. Under `jans-auth-server/client/profiles` directory, create a new directory and name it `janssen2.op.io`
 2. Copy the contents of `jans-auth/client` directory into the newly created `janssen2.op.io` directory
 3. Copy keystore file `/client/profiles/default/client_keystore.p12` from `default` directory to
    the `janssen2.op.io` directory
@@ -257,3 +257,30 @@ maven command.
 ```shell
 mvn clean compile install
 ```
+
+#### Configuring the SCIM module
+
+This module does not require a profile setup. It can be built with the below
+maven command.
+
+##### Profile setup for client and server modules
+
+Many Janssen Server modules and sub-modules use test configuration stored in a
+directory named `profile`. The profile
+directory contains files that hold important information required to run tests.
+Developers can create one or more
+profiles and use them to run tests against different Janssen Servers.
+
+Since Janssen Server has been installed with test data, the installer also
+created the profile files required to run
+the test.
+For SCIM/client module, these files are kept on the VM under
+`/opt/jans/jans-setup/output/test/scim-client` directory. Copy over this directory
+to any location on the developer workstation.
+
+Follow the steps below to configure the profile for the client module. The same
+steps should be followed for
+setting up a profile for the server module.
+
+1. Under `jans-scim/client/profiles` directory, create a new directory and name it `janssen2.op.io`
+2. Copy the contents of `scim-client/client` directory into the newly created `janssen2.op.io` directory

--- a/docs/contribute/development.md
+++ b/docs/contribute/development.md
@@ -172,18 +172,19 @@ Each module has its own set of
 tests.
 Below are the instructions for configuring each module for tests.
 
+!!! info "What is a Profile directory?"
+    - To run integration tests, the developer workspace needs to know details about he Janssen Server agaist which the tests are to be run. 
+    - Janssen Server workspace holds this information in `source-base/module/sub-module/profiles` directory.
+    - The `profiles` directory can contain one or more sub-directories, each representing a profile(i.e a target Janssen Server). These profile directories are used to hold files that contain important information required to run tests. 
+    - Developers can create one or more profiles and use them to run tests
+    against different Janssen Servers. 
+    - This guide uses Janssen Server hostname,
+     `janssen.op.io`, as profile name.
+
 #### Configuring the jans-auth-server module
 
 Configuring `jans-auth-server` module involves setting up profiles for `client`,
 `server` and `agama` sub-modules.
-
-##### Profile setup for client and server modules
-
-Many Janssen Server modules and sub-modules use test configuration stored in a 
-directory named `profile`. The profile
-directory contains files that hold important information required to run tests. 
-Developers can create one or more
-profiles and use them to run tests against different Janssen Servers. This guide uses Janssen Server host name, `janssen.op.io`, as profile name.
 
 Follow the steps below to configure the profile for the client and server sub-modules.
 
@@ -192,11 +193,12 @@ Follow the steps below to configure the profile for the client and server sub-mo
     cd source-base/jans-auth-server
    ```
 1. As a precautionary measure, let's first remove any old profile artifacts 
-from the workspace.
+from the `jans-auth-server` workspace.
    ```shell
    rm -rf ./jans-auth
    rm -rf ./client/profiles/janssen.op.io
    rm -rf ./server/profiles/janssen.op.io
+   rm -rf ./agama/engine/profiles/janssen.op.io
    ```
 
 1. Since Janssen Server has been installed with test data, the installer also
@@ -205,15 +207,17 @@ from the workspace.
    These files are kept on the VM under
    `/opt/jans/jans-setup/output/test/jans-auth` directory. Copy over `jans-auth` directory from Janssen Server VM
    to `source-base/jans-auth-server` on developer workstation.
-2. Create new profile directories for `client` and `server` sub-modules.
+2. Create new profile directories.
    ```shell
    mkdir -p ./client/profiles/janssen.op.io
    mkdir -p ./server/profiles/janssen.op.io
+   mkdir -p ./agama/engine/profiles/janssen.op.io
    ```
-2. Copy the contents of `jans-auth/client` directory into the newly created `janssen.op.io` directory
+2. Copy the contents of `jans-auth` directory into the respective sub-module's `janssen.op.io` profile directory
    ```shell
    cp ./jans-auth/client/* ./client/profiles/janssen.op.io
    cp ./jans-auth/server/* ./server/profiles/janssen.op.io
+   cp ./jans-auth/config-agama-test.properties ./agama/engine/profiles/janssen.op.io/config-agama-test.properties
    ```
 3. Copy keystore file `profiles/default/client_keystore.p12` from `default` profile directory to
    the `janssen.op.io` profile directory
@@ -221,21 +225,6 @@ from the workspace.
    cp -f ./client/profiles/default/client_keystore.p12 ./client/profiles/janssen.op.io
    cp -f ./server/profiles/default/client_keystore.p12 ./server/profiles/janssen.op.io
    ```
-
-##### Profile setup for agama sub-module
-
-Agama module code resides under `source-base/jans-auth-server/agama` directory.
-
-Follow the steps below from `agama` directory to configure the module to run the
-integration tests.
-
-1. Remove existing profile if any by deleting and recreating the directory 
-   `engine/profiles/janssen.op.io`
-2. Copy the file `jans-auth/config-agama-test.properties` to the 
-   `engine/profiles/janssen.op.io/` directory
-
-Once the above steps have been followed, the local copy of `jans-auth` directory
-that was copied from `janssen.op.io` can be deleted.
 
 ##### Running The Tests
 

--- a/docs/contribute/development.md
+++ b/docs/contribute/development.md
@@ -57,25 +57,24 @@ against an installed Janssen Server.
 ### Component Setup
 
 Instructions in this guide can be used if Janssen Server is installed on a VM. 
-Either locally on the developer workstation
-itself(as shown below) or on a remote cloud VM.
+Developers can use a virtualization software (like VMWare) or use LxD containers to create VM on developer workstation or use a remote cloud VM.
 
 ![Component Diagram](../assets/image-run-integration-test-from-workspace-06122022.png)
 
 ### Install Janssen Server
 
-Install the Janssen server using one of the methods described in the 
-installation guide. Note the points below when
+Install the Janssen server using one of the methods described in the
+VM installation guide. Note the points below when
 following installation instructions.
 
 - Make a note
   of the `host name` that you assign to the Janssen server during the 
-  installation. For this guide, the Janssen hostname would be `janssen2.op.io`
+  installation. For this guide, the Janssen hostname would be `janssen.op.io`
 - Choose to install with test data load. This can be achieved by using 
   the `-t` switch when invoking the setup script
   from installation instructions.
 
-Use the [VM installation guide](../admin/install/vm-install/README.md) for a 
+Use the [VM installation guide](../admin/install/vm-install/README.md) for the 
 complete set of instructions.
 
 Once the installation is complete, check if the `.well-known` end-points of the
@@ -87,13 +86,13 @@ from the developer's machine.
     Based on developer setup it may be necessary to add appropriate IP-HOST 
     mapping to the developer workstation. For instance, on a Linux-based 
     developer workstation, this means adding a mapping to `/etc/hosts` file. 
-    Make sure that VM's IP is mapped to a FQDN like `janssen2.op.io`. Refering 
+    Make sure that VM's IP is mapped to a FQDN like `janssen.op.io`. Refering 
     to VM with `localhost` or just IP will not work.
 
 URI for OpenID configuration `.well-known` endpoint:
 
   ```
-  https://janssen2.op.io/jans-auth/.well-known/openid-configuration
+  https://janssen.op.io/jans-auth/.well-known/openid-configuration
   ```
 
 The response received should be JSON formatted Janssen configuration details, 
@@ -102,16 +101,16 @@ similar to those below.
   ```
   {
   "request_parameter_supported" : true,
-  "pushed_authorization_request_endpoint" : "https://janssen2.op.io/jans-auth/restv1/par",
-  "introspection_endpoint" : "https://janssen2.op.io/jans-auth/restv1/introspection",
+  "pushed_authorization_request_endpoint" : "https://janssen.op.io/jans-auth/restv1/par",
+  "introspection_endpoint" : "https://janssen.op.io/jans-auth/restv1/introspection",
   "claims_parameter_supported" : false,
-  "issuer" : "https://janssen2.op.io",
+  "issuer" : "https://janssen.op.io",
   "userinfo_encryption_enc_values_supported" : [ "RSA1_5", "RSA-OAEP", "A128KW", "A256KW" ],
   "id_token_encryption_enc_values_supported" : [ "A128CBC+HS256", "A256CBC+HS512", "A128GCM", "A256GCM" ],
-  "authorization_endpoint" : "https://janssen2.op.io/jans-auth/restv1/authorize",
+  "authorization_endpoint" : "https://janssen.op.io/jans-auth/restv1/authorize",
   "service_documentation" : "http://jans.org/docs",
   "authorization_encryption_alg_values_supported" : [ "RSA1_5", "RSA-OAEP", "A128KW", "A256KW" ],
-  "id_generation_endpoint" : "https://janssen2.op.io/jans-auth/restv1/id",
+  "id_generation_endpoint" : "https://janssen.op.io/jans-auth/restv1/id",
   "claims_supported" : [ "street_address", "country", "zoneinfo", "birthdate", "role", "gender", "user_name", "formatted", "phone_mobile_number", "preferred_username", "inum", "locale", "updated_at", "post_office_box", "nickname", "preferred_language", "email", "website", "email_verified", "profile", "locality", "room_number", "phone_number_verified", "given_name", "middle_name", "picture", "name", "phone_number", "postal_code", "region", "family_name", "jansAdminUIRole" ],
   "scope_to_claims_mapping" : [ {
     "user_name" : [ "user_name" ]
@@ -141,7 +140,7 @@ needs to have the appropriate
 certificate installed. Update cacerts using the steps below:
 
 
-- extract certificate for Janssen server with name `janssen2.op.io`
+- extract certificate for Janssen server with name `janssen.op.io`
   ```
   openssl s_client -connect test.local.jans.io:443 2>&1 |sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > /tmp/httpd.crt
   ```
@@ -152,7 +151,7 @@ certificate installed. Update cacerts using the steps below:
   will prompt for cert store password. Provide
   the correct password. The default password is `changeit`.
   ```
-  keytool -import -alias janssen2.op.io -keystore /usr/lib/jvm/java-11-amazon-corretto/lib/security/cacerts -file /tmp/httpd.crt
+  keytool -import -alias janssen.op.io -keystore /usr/lib/jvm/java-11-amazon-corretto/lib/security/cacerts -file /tmp/httpd.crt
   ``` 
 
 ### Configure developer workspace
@@ -192,10 +191,10 @@ Follow the steps below to configure the profile for the client module. The same
 steps should be followed for
 setting up a profile for the server module.
 
-1. Under `jans-auth-server/client/profiles` directory, create a new directory and name it `janssen2.op.io`
-2. Copy the contents of `jans-auth/client` directory into the newly created `janssen2.op.io` directory
+1. Under `jans-auth-server/client/profiles` directory, create a new directory and name it `janssen.op.io`
+2. Copy the contents of `jans-auth/client` directory into the newly created `janssen.op.io` directory
 3. Copy keystore file `/client/profiles/default/client_keystore.p12` from `default` directory to
-   the `janssen2.op.io` directory
+   the `janssen.op.io` directory
 
 ##### Profile setup for agama sub-module
 
@@ -205,12 +204,12 @@ Follow the steps below from `agama` directory to configure the module to run the
 integration tests.
 
 1. Remove existing profile if any by deleting and recreating the directory 
-   `engine/profiles/janssen2.op.io`
+   `engine/profiles/janssen.op.io`
 2. Copy the file `jans-auth/config-agama-test.properties` to the 
-   `engine/profiles/janssen2.op.io/` directory
+   `engine/profiles/janssen.op.io/` directory
 
 Once the above steps have been followed, the local copy of `jans-auth` directory
-that was copied from `janssen2.op.io` can be deleted.
+that was copied from `janssen.op.io` can be deleted.
 
 ##### Running The Tests
 
@@ -219,7 +218,7 @@ For example, to run integration tests for `jans-auth-server` module, run the
 following maven command at the directory level:
 
   ```
-  mvn -Dcfg=janssen2.op.io test
+  mvn -Dcfg=janssen.op.io test
   ```
 
 #### Configuring the jans-core module
@@ -282,8 +281,8 @@ Follow the steps below to configure the profile for the client module. The same
 steps should be followed for
 setting up a profile for the server module.
 
-1. Under `jans-scim/client/profiles` directory, create a new directory and name it `janssen2.op.io`
-2. Copy the contents of `scim-client/client` directory into the newly created `janssen2.op.io` directory
+1. Under `jans-scim/client/profiles` directory, create a new directory and name it `janssen.op.io`
+2. Copy the contents of `scim-client/client` directory into the newly created `janssen.op.io` directory
 
 Post this, remove the `scim-client` directory.
 
@@ -291,5 +290,5 @@ Now as the profile is setup, to build the `jans-scim` module and run tests,
 use the command below:
 
   ```
-  mvn -Dcfg=janssen2.op.io test
+  mvn -Dcfg=janssen.op.io test
   ```

--- a/docs/contribute/development.md
+++ b/docs/contribute/development.md
@@ -61,6 +61,9 @@ Developers can use a virtualization software (like VMWare) or use LxD containers
 
 ![Component Diagram](../assets/image-run-integration-test-from-workspace-06122022.png)
 
+!!! note "OS platform for Developer Workstation"
+    Steps in this guide are applicable to any OS platform a Developer workstation may have. Example commands given in this guide are for Ubuntu Linux based workstation.  
+
 ### Install Janssen Server
 
 Install the Janssen server using one of the methods described in the
@@ -141,7 +144,7 @@ certificate installed. Update cacerts using the steps below:
 
 
 - extract certificate for Janssen server with name `janssen.op.io`
-  ```
+  ```shell title="On Developer Workstation"
   openssl s_client -connect test.local.jans.io:443 2>&1 |sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > /tmp/httpd.crt
   ```
   this command takes a few seconds to return.
@@ -161,9 +164,11 @@ developer workstation as well, we will
 create and configure the code base on the developer workspace to run integration
 tests.
 
-Get Janssen server code from [Janssen GitHub repository](https://github.com/JanssenProject/jans).
+Get Janssen server code from [Janssen GitHub repository](https://github.com/JanssenProject/jans). Note the path to this location, we will refer to it as `source-base`.
 
-Janssen Server is composed of multiple modules. Each module has its own set of 
+Janssen Server is composed of multiple modules. For example `source-base/jans-auth-server`, `source-base/jans-link` etc. 
+
+Each module has its own set of 
 tests.
 Below are the instructions for configuring each module for tests.
 
@@ -178,27 +183,48 @@ Many Janssen Server modules and sub-modules use test configuration stored in a
 directory named `profile`. The profile
 directory contains files that hold important information required to run tests. 
 Developers can create one or more
-profiles and use them to run tests against different Janssen Servers.
+profiles and use them to run tests against different Janssen Servers. This guide uses Janssen Server host name, `janssen.op.io`, as profile name.
 
-Since Janssen Server has been installed with test data, the installer also 
-created the profile files required to run
-the test.
-These files are kept on the VM under 
-`/opt/jans/jans-setup/output/test/jans-auth` directory. Copy over this directory
-to any location on the developer workstation.
+Follow the steps below to configure the profile for the client and server sub-modules.
 
-Follow the steps below to configure the profile for the client module. The same 
-steps should be followed for
-setting up a profile for the server module.
+1. Move to the module directory
+   ```shell
+    cd source-base/jans-auth-server
+   ```
+1. As a precautionary measure, let's first remove any old profile artifacts 
+from the workspace.
+   ```shell
+   rm -rf ./jans-auth
+   rm -rf ./client/profiles/janssen.op.io
+   rm -rf ./server/profiles/janssen.op.io
+   ```
 
-1. Under `jans-auth-server/client/profiles` directory, create a new directory and name it `janssen.op.io`
+1. Since Janssen Server has been installed with test data, the installer also
+   created the profile files required to run
+   the test.
+   These files are kept on the VM under
+   `/opt/jans/jans-setup/output/test/jans-auth` directory. Copy over `jans-auth` directory from Janssen Server VM
+   to `source-base/jans-auth-server` on developer workstation.
+2. Create new profile directories for `client` and `server` sub-modules.
+   ```shell
+   mkdir -p ./client/profiles/janssen.op.io
+   mkdir -p ./server/profiles/janssen.op.io
+   ```
 2. Copy the contents of `jans-auth/client` directory into the newly created `janssen.op.io` directory
-3. Copy keystore file `/client/profiles/default/client_keystore.p12` from `default` directory to
-   the `janssen.op.io` directory
+   ```shell
+   cp ./jans-auth/client/* ./client/profiles/janssen.op.io
+   cp ./jans-auth/server/* ./server/profiles/janssen.op.io
+   ```
+3. Copy keystore file `profiles/default/client_keystore.p12` from `default` profile directory to
+   the `janssen.op.io` profile directory
+   ```shell
+   cp -f ./client/profiles/default/client_keystore.p12 ./client/profiles/janssen.op.io
+   cp -f ./server/profiles/default/client_keystore.p12 ./server/profiles/janssen.op.io
+   ```
 
 ##### Profile setup for agama sub-module
 
-Agama module code resides under `jans/agama` directory.
+Agama module code resides under `source-base/jans-auth-server/agama` directory.
 
 Follow the steps below from `agama` directory to configure the module to run the
 integration tests.

